cpp/src/arrow/dataset/file_base.h

```diff
@@ -211,8 +221,12 @@ class ARROW_DS_EXPORT FileFragment : public Fragment,
 
   FileSource source_;
   std::shared_ptr<FileFormat> format_;
-
   friend class FileFormat;
+
+  // we do not want ParquetFileFragment to inherit these things.
+ private:
+  int64_t start_byte_ = 0;
+  int64_t end_byte_ = 0;
```

Comment:
Does this mean an empty range? It seems a bit confusing otherwise (should `util::optional` be used perhaps?)

Reply:
Or, better, perhaps `util::optional<ReadRange>`.

Reply:
Will use util::optional<ReadRange>